core/src/main/scala/org/apache/spark/ExecutorAllocationManager.scala

```diff
@@ -262,15 +267,22 @@ private[spark] class ExecutorAllocationManager(
     val maxNeeded = maxNumExecutorsNeeded
 
     if (maxNeeded < numExecutorsTarget) {
-      // The target number exceeds the number we actually need, so stop adding new
-      // executors and inform the cluster manager to cancel the extra pending requests
-      val oldNumExecutorsTarget = numExecutorsTarget
-      numExecutorsTarget = math.max(maxNeeded, minNumExecutors)
-      client.requestTotalExecutors(numExecutorsTarget)
-      numExecutorsToAdd = 1
-      logInfo(s"Lowering target number of executors to $numExecutorsTarget because " +
-        s"not all requests are actually needed (previously $oldNumExecutorsTarget)")
-      numExecutorsTarget - oldNumExecutorsTarget
+      if (!numTargetExecutorAdjustable.get) {
+        // Keep the initial number of target executor to not ramp down until the first job is
```

Comment:
That is the intended behavior.  My complaint is that this patch makes it so that, in addition to that behavior, during the initialization period we can't ramp our target down at all, even if the new target would stay above the initial number of executors.  Does that make sense?